player/command.c

```diff
@@ -3558,6 +3563,103 @@ static int mp_property_script_props(void *ctx, struct m_property *prop,
     return M_PROPERTY_NOT_IMPLEMENTED;
 }
 
+static int do_list_user_data(int item, int action, void *arg, void *ctx);
+
+struct user_data_ctx {
+    MPContext *mpctx;
+    struct m_property *prop;
+    mpv_node *node;
+};
+
+static int do_op_user_data(struct user_data_ctx* ctx, int action, void *arg)
+{
+    MPContext *mpctx = ctx->mpctx;
+    struct m_property *prop = ctx->prop;
+    mpv_node *node = ctx->node;
+
+    if (node && node->format == MPV_FORMAT_NODE_ARRAY &&
+        (action == M_PROPERTY_KEY_ACTION || action == M_PROPERTY_GET ||
+         action == M_PROPERTY_GET_NODE))
+        return m_property_read_list(action, arg, node->u.list->num, &do_list_user_data, ctx);
```

Comment:
A few more comments around here would be nice too.

Reply:
Added.